### PR TITLE
add clustertag to machine if it is missing

### PR DESCRIPTION
**Pull request by mwennrich**
resolves https://github.com/metal-stack/metal-ccm/issues/77

### CODE REVIEW

**Comment by mwennrich** [comment]
```
cloud-controller-manager I0714 05:38:31.978020       1 tags.go:81] added cluster tag "60baa976-a9c1-4e50-916a-a2618c3eefb0" to machine "00000000-0000-0000-0000-ac1f6bd390a6"
```

File: pkg/tags/tagUtils.go [hunk 1]
```diff
@@ -37,3 +38,15 @@ func IsMemberOfCluster(tag, clusterID string) bool {
 	}
 	return false
 }
+
+func GetMachineClusterTag(tags []string) (string, bool) {
```
Comment (majst01) [inline pkg/tags/tagUtils.go:42]: What if clustertag is present but values is different or missing ?

File: pkg/resources/metal/machine.go [hunk 2]
```diff
@@ -133,6 +133,22 @@ func (ms *MetalService) GetMachineFromNode(ctx context.Context, node *v1.Node) (
 	return machine, err
 }
 
+// UpdateMachineTags sets the machine tags.
+func (ms *MetalService) UpdateMachineTags(ctx context.Context, m *string, tags []string) error {
+	if m == nil {
+		return fmt.Errorf("machine is nil")
+	}
+
+	_, err := ms.client.Machine().UpdateMachine(machine.NewUpdateMachineParams().WithBody(&models.V1MachineUpdateRequest{
```
Comment (Gerrit91) [inline pkg/resources/metal/machine.go:142]: Misses using `.WithContext(ctx)`

File: pkg/tags/tagUtils.go (hunk 1 above)
Comment (Gerrit91) [inline pkg/tags/tagUtils.go:42]: You can also consider tag helpers from the metal-lib, already there and tested. :)

https://github.com/metal-stack/metal-lib/blob/master/pkg/tag/helpers.go

Comment (mwennrich) [reply pkg/tags/tagUtils.go:42]: Don't know either. Overwriting a clustertag seems a bit scary.

Comment (Gerrit91) [reply pkg/tags/tagUtils.go:42]: IMHP the metal-ccm knows for sure which cluster ID is correct and "owns" this flag. So it should also correct the value if it's different.

**Review by majst01 — APPROVED** [review]

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**